modules/fedimint-mint/src/lib.rs

```diff
@@ -356,7 +356,7 @@ impl ServerModulePlugin for Mint {
     type ConsensusItem = MintOutputConfirmation;
     type VerificationCache = VerificationCache;
 
-    fn module_key(&self) -> ModuleKey {
+    fn module_key(&self) -> ModuleInstanceId {
```

Comment:
I think for this PR to work every type needs to carry around the instance it belongs to, otherwise adding the same module twice will lead to collisions/not being able to properly dispatch inputs and outputs.

Reply:
The consensus items (inputs, outputs, etc.) are already encoded with `ModuleKey` (now `ModuleInstanceId`) prefix, so they uniquely identify the instance.